tests/integration_tests/lightgbm_tuner_tests/test_optimize.py

```diff
@@ -0,0 +1,408 @@
+from optuna.integration.lightgbm_tuner.optimize import _TimeKeeper
+from optuna.integration.lightgbm_tuner.optimize import _timer
+from optuna.integration.lightgbm_tuner.optimize import BaseTuner
+from optuna.integration.lightgbm_tuner.optimize import OptunaObjective
+from optuna import type_checking
+
+
+if type_checking.TYPE_CHECKING:
+    from type_checking import Any  # NOQA
+    from type_checking import Dict  # NOQA
+    from type_checking import Generator  # NOQA
+    from type_checking import List  # NOQA
+    from type_checking import Tuple  # NOQA
+
+if sys.version_info >= (3, 5):
+    from contextlib import ExitStack
+
+
+@contextlib.contextmanager
+def turnoff_tuner():
+    # type: () -> Generator[None, None, None]
+
+    fqn_prefix = 'optuna.integration.lightgbm_tuner.LightGBMTuner'
+    mock_pairs = [
+        (fqn_prefix + '.__init__', None),
+        (fqn_prefix + '.run', True),
+        (fqn_prefix + '._parse_args', None),
+        (fqn_prefix + '.get_params', {}),
+    ]  # type: List[Tuple[str, Any]]
+
+    if sys.version_info >= (3, 5):
+        with ExitStack() as stack:
+            for fqn, return_value in mock_pairs:
+                stack.enter_context(mock.patch(fqn, return_value=return_value))
+            yield
+
+
+@contextlib.contextmanager
+def turnoff_train():
+    # type: () -> Generator[None, None, None]
+
+    unexpected_value = 0.5
+    dummy_num_iterations = 1234
+
+    class DummyBooster(object):
+        def __init__(self):
+            # type: () -> None
+
+            self.best_score = {
+                'valid_0': {'binary_logloss': unexpected_value},
+            }
+
+        def current_iteration(self):
+            # type: () -> int
+
+            return dummy_num_iterations
+
+    dummy_booster = DummyBooster()
+
+    with mock.patch('lightgbm.train', return_value=dummy_booster):
+        yield
+
+
+class TestOptunaObjective(object):
+
+    def test_init_(self):
+        # type: () -> None
+
+        target_param_names = ['learning_rate']  # Invalid parameter name.
+
+        with pytest.raises(NotImplementedError) as execinfo:
+            OptunaObjective(target_param_names, {}, None, {}, 0)
+
+        assert execinfo.type is NotImplementedError
+
+    def test_call(self):
+        # type: () -> None
+
+        target_param_names = ['lambda_l1']
+        lgbm_params = {}  # type: Dict[str, Any]
+        train_set = lgb.Dataset(None)
+        val_set = lgb.Dataset(None)
+
+        lgbm_kwargs = {'valid_sets': val_set}
+        best_score = -np.inf
+
+        with turnoff_train():
+            objective = OptunaObjective(
+                target_param_names,
+                lgbm_params,
+                train_set,
+                lgbm_kwargs,
+                best_score,
+            )
+            study = optuna.create_study(direction='minimize')
+            study.optimize(objective, n_trials=10)
+
+            assert study.best_value == 0.5
+
+
+class Test_TimeKeeper(object):
```

Comment:
`TestTimeKeeper` would be more consistent with other test classes.

Reply:
Ok. Fixed: ec045b0